src/application.ts

```diff
@@ -145,15 +151,81 @@ export default class Application {
           processor.appInstance,
         )) as OperationProcessor<Resource>;
         const result = await relatedProcessor.execute(deserializedOperation);
-
-        return this.buildOperationResponse(result, processor.appInstance);
+        if (result) {
+          await this.injectResourceMeta(result, processor, op);
+        }
+        return this.buildOperationResponse(result, processor, op);
       }
     }
 
     const deserializedOperation = await this.serializer.deserializeResource(op, resourceClass);
     const result = await processor.execute(deserializedOperation);
 
-    return this.buildOperationResponse(result, processor.appInstance);
+    if (result) {
+      await this.injectResourceMeta(result, processor, op);
+    }
+
+    return this.buildOperationResponse(result, processor, op);
+  }
+
+  async injectResourceMeta(
+    result: Resource | Resource[],
+    processor: OperationProcessor<Resource>,
+    op: Operation,
+  ): Promise<void> {
+    if (OPERATIONS_INCOMPATIBLE_WITH_META_INJECTION.includes(op.op)) {
+      return;
+    }
+
+    const resourceMetaHookToCallForOperation = `resourceMetaFor${classify(op.op)}`;
+    if (Array.isArray(result)) {
+      for (const resource of result) {
+        const resourceMeta = await processor.resourceMeta(resource);
+        const resourceMetaFor = await processor.resourceMetaFor(op, resource);
+        const resourceMetaForOp = await processor[resourceMetaHookToCallForOperation](resource);
+        resource.meta = {
+          ...resourceMetaForOp,
+          ...resourceMetaFor,
+          ...resourceMeta,
+        } as Meta;
+        if (isEmptyObject(resource.meta)) {
+          delete resource.meta;
+        }
+      }
+    } else {
+      const resourceMeta = await processor.resourceMeta(result);
+      const resourceMetaFor = await processor.resourceMetaFor(op, result);
+      const resourceMetaForOp = await processor[resourceMetaHookToCallForOperation](result);
+      result.meta = {
+        ...resourceMetaForOp,
+        ...resourceMetaFor,
+        ...resourceMeta,
+      } as Meta;
+      if (isEmptyObject(result.meta)) {
+        delete result.meta;
+      }
+    }
+  }
+
+  async buildDocumentMeta(
+    data: Resource | Resource[] | void,
+    processor: OperationProcessor<Resource>,
+    op: Operation,
+  ): Promise<MaybeMeta> {
+    if (!data) {
```

Comment:
Can we please add the `OPERATIONS_INCOMPATIBLE_WITH_META_INJECTION` return here as well? We call `buildDocumentMeta` every time we want to `buildOperationResponse` (https://github.com/kurierjs/kurier/pull/327/files#diff-16a8f18a1e6d24ab6125db9b965edc8d7970d184e44ada35f3ccbf71a21670a4R320)

Reply:
Thanks a lot @joelalejandro 🎉 ✨